### ISSUE TITLE
Feature request: Let me pass my own iframe element (or atts) instead of just the src.

### ISSUE THREAD

**Issue by nextgenthemes** [henrygd/bigger-picture#45]
First, thanks @henrygd! I really like the evolution from big-picture to bigger-picture.

I like to add a lot of custom iframe attributes and ran into this issue: 

I currently do this in `onOpen`:

```
function copyIframeAttr( source: HTMLIFrameElement, target: HTMLIFrameElement ) {
	target.dataset.arve = source.dataset.arve;
	target.allow = source.allow;
	target.title = source.title;
	source.sandbox.forEach( ( token ) => {
		target.sandbox.add( token );
	} );
}
```

I copy some attributes from an Iframe I have prepared in PHP `target` is the `iframe` that BP created at that point. However, I just found out it's not working (popups that `sandbox` is supposed to prevent just open anyway). It's correctly added to the HTML if I inspect it, but specifically `sandbox` has no effect. I think it's because the iframe already starts loading. My guess is things like `credentialless` may also fail, but I have not tried it. I am about to make my iframes even more private, and that is getting [pretty insane these days.](https://dustri.org/b/youtube-video-embedding-harm-reduction.html)

So my guess is the easiest solution would be to pass a complete iframe element and let PB put the style, class needed on it and this `loading` callback  / Swelte magic. I do not understand it, maybe you have a solution for me. Like, could I completely swap the iframe out with my own and add the `loading` callback back on my iframe?

Or some kind of hook that would let people add their own attributes to the iframe **before** the `src` is added, and it starts loading, because I think that is the issue here. And just in general, I would like to add attributes in a more convenient way: I think it would be beneficial for not only me. [Because of this](https://dustri.org/b/youtube-video-embedding-harm-reduction.html), for example.

I only had a few baby steps in Swelte not very good with JS anyway, if I would know how, I would do a PR for this myself, but I don't.

**Comment by henrygd** [comment]
@nextgenthemes

Just released 1.1.16 on npm that adds the [`attr`](https://github.com/henrygd/bigger-picture/blob/main/readme.md#attr) option.

Can you test that out and let me know if it solves the problem?

**Comment by nextgenthemes** [comment]
@henrygd Thank you! I tested it with iframe and video not images so far.

For the iframe I can not pass the `src` with `data-attr` in html and have to explicitly set `data-iframe` but that is fine with me.
